custom/icds/views/custom_sms_report.py

```diff
@@ -22,22 +25,54 @@ def dispatch(self, *args, **kwargs):
 
     @property
     def page_context(self):
-        self.request_form = CustomSMSReportRequestForm()
+        report_count = len(self._get_active_reports())
+        disable_submit = report_count >= 3
+        if disable_submit:
+            messages.error(self.request, 'Only 3 concurrent reports are allowed at a time.\
+                Please wait for them to finish.')
+        if report_count > 0:
+            display_message = self._prepare_display_message()
+            messages.info(self.request, display_message)
+        self.request_form = CustomSMSReportRequestForm(disable_submit=disable_submit)
         return {
             'request_form': self.request_form,
+            'disable_submit': disable_submit,
         }
 
+    @memoized
+    def _get_active_reports(self):
+        report_tracker = CustomSMSReportTracker(self.request.domain)
+        return report_tracker.active_reports
+
+    def _prepare_display_message(self):
+        reports_in_progress = self._get_active_reports()
+        report_count = len(reports_in_progress)
+        message = _('Reports in progress: ')
+        for index, report in enumerate(reports_in_progress):
+            message += '"{report_id}"'.format(report_id=report)
+            if index != report_count - 1:
+                message += ', '
+        return message
+
     def post(self, request, *args, **kwargs):
+        report_tracker = CustomSMSReportTracker(request.domain)
+        reports_in_progress = report_tracker.active_reports
+        report_count = len(reports_in_progress)
+
         self.request_form = CustomSMSReportRequestForm(request.POST)
-        user_email = request.user.email
-        if not user_email:
-            messages.error(self.request, _("Unable to find any email associated with your account"))
-            return self.get(*args, **kwargs)
-        if self.request_form.is_valid():
+
+        if self.request_form.is_valid() and report_count < 3:
```

Comment:
this logic of checking with 3 would look better at one place, i think on tracker itself and then you can just call it
`tracker.can_add_new_report` or `tracker.max_limit_reached` or something better.

Reply:
Added a property `max_report_count` in the util class to take care of it.